contracts/contracts/coordination/ManagedAllowList.sol

```diff
@@ -2,149 +2,97 @@
- * The Subscription contract is used to enforce limits on the number of authorization actions that can be performed, and these limits can be set and updated through the ManagedAllowList contract.
  */
-contract ManagedAllowList is GlobalAllowList {
-    mapping(bytes32 => uint256) internal allowance;
+contract ManagedAllowList is GlobalAllowList, AccessControlUpgradeable {
+    mapping(address authAdmin => mapping(bytes32 lookupKey => bool)) public authAdmins;
 
-    /**
-     * @notice The Subscription contract used to manage authorization caps
-     */
-    // TODO: Should it be updatable?
-    UpfrontSubscriptionWithEncryptorsCap public immutable subscription;
-
-    /**
-     * @notice Emitted when an administrator cap is set
-     * @param ritualId The ID of the ritual
-     * @param _address The address of the administrator
-     * @param cap The cap value
-     */
-    event AdministratorCapSet(uint32 indexed ritualId, address indexed _address, uint256 cap);
+    bytes32 public constant COHORT_ADMIN_BASE = keccak256("COHORT_ADMIN");
+    bytes32 public constant AUTH_ADMIN_BASE = keccak256("AUTH_ADMIN");
 
     /**
-     * @notice Sets the coordinator and subscription contracts
-     * @dev The coordinator and subscription contracts cannot be zero addresses
+     * @notice Sets the coordinator contract
+     * @dev The coordinator contract cannot be a zero address and must have a valid number of rituals
      * @param _coordinator The address of the coordinator contract
-     * @param _subscription The address of the subscription contract
      */
-    constructor(
-        Coordinator _coordinator,
-        UpfrontSubscriptionWithEncryptorsCap _subscription // TODO replace with IFeeModel subscription
-    ) GlobalAllowList(_coordinator) {
-        require(address(_subscription) != address(0), "Subscription cannot be the zero address");
-        subscription = _subscription;
+    constructor(Coordinator _coordinator) GlobalAllowList(_coordinator) {
+        _disableInitializers();
     }
 
-    /**
-     * @notice Checks if the sender is the authority of the ritual
-     * @param ritualId The ID of the ritual
-     */
-    modifier onlyCohortAuthority(uint32 ritualId) {
-        require(
-            coordinator.getAuthority(ritualId) == msg.sender,
-            "Only cohort authority is permitted"
-        );
-        _;
-    }
-
-    /**
-     * @notice Checks if the sender has allowance to set authorizations
-     * @dev This function overrides the canSetAuthorizations modifier in the GlobalAllowList contract
-     * @param ritualId The ID of the ritual
-     */
-    modifier canSetAuthorizations(uint32 ritualId) override {
-        require(getAllowance(ritualId, msg.sender) > 0, "Only administrator is permitted");
-        _;
+    function ritualRole(uint32 ritualId, bytes32 role) public view returns (bytes32) {
+        return keccak256(abi.encodePacked(ritualId, role));
```

Comment:
Awesome idea 💯 Let's go a step further and have two functions based on this:
```js
function cohortAdminRole(uint32 ritualId) public view returns (bytes32);
function authAdminRole(uint32 ritualId) public view returns (bytes32);
```
Also, can this function be `pure`? At first sight, I don't see why not 🤔

Reply:
good idea!
✔️